app/models/storage_location.rb

```diff
@@ -205,6 +206,38 @@ def decrease_inventory(itemizable_array)
     log
   end
 
+  def manage_inventory_out(line_items_array, kit_id, operation, adjust_by)
+    allocation = allocations.find_or_create_by!(kit_id: kit_id, organization_id: organization.id, allocation_type: "inventory_out")
+    adjust_by  = operation == 'allocate' ? -1 : 1
+    line_items = allocation.line_items
+    if line_items.present?
+      line_items_array.each do |line_item|
+        line_item_record = line_items.find_by(item_id: line_item[:item_id])
+        new_quantity =  line_item_record[:quantity] + line_item[:quantity] * adjust_by
+        line_item_record.update!(quantity: new_quantity)
+      end
+    else # operation will be always allocate
+      line_items_array.each do |line_item|
+        allocation.line_items.create!(item_id: line_item[:item_id], quantity: line_item[:quantity].to_i * adjust_by)
+      end
+    end
+  end
+
+  def manage_inventory_in(associated_kit, kit_id, operation)
+    allocation = allocations.find_or_create_by!(kit_id: kit_id, organization_id: organization.id, allocation_type: "inventory_in")
+    adjust_by = associated_kit[0][:quantity]
```

Comment:
Why is `associated_kit` an array of hashes? That's not indicative of the argument name.

Reply:
change applied